test/e2e/proxy_test.go

```diff
@@ -150,16 +180,100 @@ func TestProxyFlow(t *testing.T) {
 					require.NoError(t, err)
 					assert.NotEmpty(t, found)
 
+					newApp := user.getApplication(t, proxyCl, applicationName)
+					assert.NotEmpty(t, newApp)
+
 					// Double check that the Application does exist using a regular client (non-proxy)
-					namespacedName := types.NamespacedName{Namespace: tenantNsName(user.compliantUsername), Name: applicationName}
-					createdApp := &appstudiov1.Application{}
-					err = user.expectedMemberCluster.Client.Get(context.TODO(), namespacedName, createdApp)
-					require.NoError(t, err)
-					require.NotEmpty(t, createdApp)
+					createdApp := user.getApplicationWithoutProxy(t, applicationName)
 					assert.Equal(t, expectedApp.Spec.DisplayName, createdApp.Spec.DisplayName)
 				}
 			})
 
+			t.Run("use proxy to update a HAS Application CR in the user appstudio namespace via proxy API", func(t *testing.T) {
```

Comment:
this test (and also the following ones) depends on the execution of the first one, so they should be implemented as sub-tests of the first one, and not at the same level.

Reply:
Good point. It should be resolved now. Thank you!